### DISCUSSION TITLE
Extremely Low FPS in Manual Control

### DISCUSSION THREAD

**Discussion by malak-peter** [carla-simulator/carla#4691] [Q&A]
Hello everyone,
I'm working on a project in CARLA 0.9.10, and I have to run manual_control.py to test my codes.
When I run manual_control.py, the server has constant 3 FPS, while the Client has 60 FPS. The simulation lags and reacts slowly to the key commands (WASD).

![image](https://user-images.githubusercontent.com/85319341/135310211-9b0436c8-2ebf-423c-a7c9-9ba3024f8d9c.png)

It worked properly a few weeks ago, I do not know what changed.
Do you have any idea, why this is happening and how can I solve it?

Thank you in advance!

**Comment by iammix** [comment]
GPU might be a reason. A simple test is to run the same script in Town02.

**Reply by malak-peter** [reply]
Thanks for your reply @iammix!
The same problem occurred when I run manual_control.py in Town02.
Do you have any other idea?
If GPU is the reason, why could I run it properly few weeks ago?

**Reply by iammix** [reply]
Another idea is to install CARLA 0.9.12. Maybe there are some improvements that might fix your issue.
I faced the same problem a long time ago, but after a PC restart, the problem was solved.

**Comment by HertzDav** [comment]
Hello！
You should find the setting of "background frame rate reduction" in UE4 and turn it off to solve it!

**Comment by tian9667** [comment]
Hello Everyone~
Open your Unreal Engine -> Edit -> Editor Preferens... -> General -> Performance -> "Use Less CPU when in Background" (turn it off)      (^_^ I'm Taiwan Handsome boy~ WOW HAHAHAHA)

**Reply by alex880317** [reply]
Thanks for your suggestion
I have the same problem in manual_control.py
After setting it up according to your way. I get 30~40 fps on carla server.
btw I'm from taiwan too.(goodjob Taiwan handsome boy)
